src/components/dialogHelper/dialogHelper.js

```diff
@@ -86,10 +90,10 @@ import '../../assets/css/scrollstyles.scss';
                 document.body.classList.remove('noScroll');
             }
 
-            if (!self.closedByBack && isHistoryEnabled(dlg)) {
-                const state = window.history.state || {};
-                if (state.dialogId === hash) {
-                    appRouter.back();
+            if (isHistoryEnabled(dlg)) {
+                const state = history.location.state || {};
+                if (state.dialogs?.length > 0 && state.dialogs[0] === hash) {
```

Comment:
Maybe need the last one?

Reply:
Yeah this was wrong. I was pushing new dialogs to the end of the array, but checking the first entry. Should be fixed now.